### ISSUE TITLE
Evaluation failed: TypeError: msg.quotedMsgObj is not a function

### ISSUE THREAD

**Issue by RC047** [pedroslopez/whatsapp-web.js#1835]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Error: Evaluation failed: TypeError: msg.quotedMsgObj is not a function
    at __puppeteer_evaluation_script__:3:24
    at ExecutionContext._evaluateInternal (C:\Users\rendy\OneDrive\Desktop\kuhong-bot-v3\node_modules\puppeteer\lib\cjs\puppeteer\common\ExecutionContext.js:221:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ExecutionContext.evaluate (C:\Users\rendy\OneDrive\Desktop\kuhong-bot-v3\node_modules\puppeteer\lib\cjs\puppeteer\common\ExecutionContext.js:110:16)
    at async Message.getQuotedMessage (C:\Users\rendy\OneDrive\Desktop\kuhong-bot-v3\node_modules\whatsapp-web.js\src\structures\Message.js:307:27)
    at async Object.handler (C:\Users\rendy\OneDrive\Desktop\kuhong-bot-v3\main.js:1748:18)

### Expected behavior

-

### Steps to Reproduce the Bug or Issue

When the ```<Message>.getQuotedMessage()``` function called

### Relevant Code

-

### Browser Type

Google Chrome

### WhatsApp Account Type

Standard

### Does your WhatsApp account have multidevice enabled?

Yes, I am using Multi Device

### Environment

OS: Windows 10 Enterprise
whatsapp-web.js version: 1.18.3 (latest)
WhatsApp Web Version: 2.2244.6
Node.js Version: 16.17.1

**Comment by jeduoliveira** [comment]
I got same error :(

**Comment by n0buholic** [comment]
same error

**Comment by anderson-scherdovski** [comment]
here too

**Comment by jimit24365** [comment]
workaround is to clearing .wwebjs_auth/ and re-linking via QR-CODE every time I encounter this.

**Comment by jeduoliveira** [comment]
> workaround is to clearing .wwebjs_auth/ and re-linking via QR-CODE every time I encounter this.

did not work!

**Comment by smilingOrange** [comment]
https://github.com/pedroslopez/whatsapp-web.js/blob/a60f5ebf407e4f8183947c497e65bb149a707dc7/src/structures/Message.js#L309

Change this line to:-
return msg.getRawQuotedMsgObj().serialize()

**Comment by jeduoliveira** [comment]
> https://github.com/pedroslopez/whatsapp-web.js/blob/a60f5ebf407e4f8183947c497e65bb149a707dc7/src/structures/Message.js#L309
> 
> Change this line to:- return msg.getRawQuotedMsgObj().serialize()

It's working!!! Thanks!!!

**Comment by Mohammed-800** [comment]
> > https://github.com/pedroslopez/whatsapp-web.js/blob/a60f5ebf407e4f8183947c497e65bb149a707dc7/src/structures/Message.js#L309
> > 
> > Change this line to:- return msg.getRawQuotedMsgObj().serialize()
> 
> It's working!!! Thanks!!!

after That has error 

Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'serialize')
    at _puppeteer_evaluation_script_:3:44

**Comment by Mohammed-800** [comment]
> https://github.com/pedroslopez/whatsapp-web.js/blob/a60f5ebf407e4f8183947c497e65bb149a707dc7/src/structures/Message.js#L309
> 
> Change this line to:- return msg.getRawQuotedMsgObj().serialize()

has error agin

Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'serialize')
    at _puppeteer_evaluation_script_:3:44



Why?

**Comment by tofers** [comment]
basic  =  quotedMsgObj     |  not function getRawQuotedMsgObj 
business  = getRawQuotedMsgObj  | not function  quotedMsgObj

**Comment by tofers** [comment]
basic and business  

work option msg['_quotedMsgObj'].serialize()

**Comment by Agusteando** [comment]
None of the changes proposed here are fixing the problem

**Comment by smilingOrange** [comment]
> 

I'm using standard Whatsapp and npm@6.14.13

**Comment by smilingOrange** [comment]
> None of the changes proposed here are fixing the problem

Please give details

**Comment by RC047** [comment]
> basic and business
> 
> work option msg['_quotedMsgObj'].serialize()
> 
> let quotedMsgObj = ('quotedMsgObj' in msg) ? msg.quotedMsgObj() : msg.getRawQuotedMsgObj();
> 
> return quotedMsgObj.serialize();

Why not worked for me

**Comment by stefanfuchs** [comment]
This could solve it. Adding some ifs to check for undefined messages.
I'll test soon if this solution works.

```js
async getQuotedMessage() {
        if (!this.hasQuotedMsg) return undefined;

        const quotedMsg = await this.client.pupPage.evaluate((msgId) => {
            let msg = window.Store.Msg.get(msgId);
            let quotedMsgObj = ('quotedMsgObj' in msg) ? msg.quotedMsgObj() : msg.getRawQuotedMsgObj();
            if (!quotedMsgObj) return undefined;

            return quotedMsgObj.serialize();
        }, this.id._serialized);

        if (!quotedMsg) return undefined;
        return new Message(this.client, quotedMsg);
    }
```

**Comment by stefanfuchs** [comment]
Seems to work now with my solution.

**Comment by marcelofp** [comment]
> Seems to work now with my solution.

it doesn't work, I did the test and now it's giving problems in other functions, for example

`const mediaData = hasQuotedMsg ? (quotedMsg.hasMedia ? await quotedMsg.downloadMedia() : undefined) : (hasMedia ? await msg.downloadMedia() : undefined)`

I get this error:
TypeError: Cannot read property 'hasMedia' of undefined

**Comment by smilingOrange** [comment]
> > Seems to work now with my solution.
> 
> it doesn't work, I did the test and now it's giving problems in other functions, for example
> 
> `const mediaData = hasQuotedMsg ? (quotedMsg.hasMedia ? await quotedMsg.downloadMedia() : undefined) : (hasMedia ? await msg.downloadMedia() : undefined)`
> 
> I get this error: TypeError: Cannot read property 'hasMedia' of undefined

The query for the quoted message will only return the quoted message if the original message is relatively recent.

**Comment by smilingOrange** [comment]
> > Seems to work now with my solution.
> 
> it doesn't work, I did the test and now it's giving problems in other functions, for example
> 
> `const mediaData = hasQuotedMsg ? (quotedMsg.hasMedia ? await quotedMsg.downloadMedia() : undefined) : (hasMedia ? await msg.downloadMedia() : undefined)`
> 
> I get this error: TypeError: Cannot read property 'hasMedia' of undefined

Please post the entire error message.

**Comment by pedroslopez** [comment]
Thanks for reporting, this issue has now been fixed in v1.18.4.
